File: shapeView/src/main/java/com/jingzz/jzwidget/helper/IShapeTextHelper.java
package com.jingzz.jzwidget.helper;

import android.view.MotionEvent;

public interface IShapeTextHelper{
    boolean onTouchEvent(MotionEvent ev);
    void shapeCreate();

    void setDrawableTopWidth(int drawableTopWidth);
    void setDrawableLeftWidth(int drawableLeftWidth);
    void setDrawableRightWidth(int drawableRightWidth);
    void setDrawableBottomWidth(int drawableBottomWidth);

    void setDrawableTopHeight(int drawableTopHeight);
    void setDrawableLeftHeight(int drawableLefHeight);
    void setDrawableRightHeight(int drawableRightHeight);
    void setDrawableBottomHeight(int drawableBottomHeight);

    int getDrawableTopWidth();
    int getDrawableLeftWidth();
    int getDrawableRightWidth();
    int getDrawableBottomWidth();

    int getDrawableTopHeight();
    int getDrawableLeftHeight();
    int getDrawableRightHeight();
    int getDrawableBottomHeight();

}

File: shapeView/src/main/java/com/jingzz/shapeView/helper/IShapeTextHelper.kt
package com.jingzz.shapeView.helper

import android.view.MotionEvent




interface IShapeTextHelper<T> {
    fun onTouchEvent(ev: MotionEvent): Boolean
    fun shapeCreate()

    fun setDrawableTopWidth(drawableTopWidth: Int):T
    fun setDrawableLeftWidth(drawableLeftWidth: Int):T
    fun setDrawableRightWidth(drawableRightWidth: Int):T
    fun setDrawableBottomWidth(drawableBottomWidth: Int):T

    fun setDrawableTopHeight(drawableTopHeight: Int):T
    fun setDrawableLeftHeight(drawableLefHeight: Int):T
    fun setDrawableRightHeight(drawableRightHeight: Int):T
    fun setDrawableBottomHeight(drawableBottomHeight: Int):T

    fun getDrawableTopWidth(): Int
    fun getDrawableLeftWidth(): Int
    fun getDrawableRightWidth(): Int
    fun getDrawableBottomWidth(): Int

    fun getDrawableTopHeight(): Int
    fun getDrawableLeftHeight(): Int
    fun getDrawableRightHeight(): Int
    fun getDrawableBottomHeight(): Int
}